client/tx/search.go

```diff
@@ -139,45 +139,45 @@ func FormatTxResults(cdc *codec.Codec, res []*ctypes.ResultTx) ([]Info, error) {
 // Search Tx REST Handler
 func SearchTxRequestHandlerFn(cliCtx context.CLIContext, cdc *codec.Codec) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		tag := r.FormValue("tag")
-		if tag == "" {
-			w.WriteHeader(http.StatusBadRequest)
-			w.Write([]byte("You need to provide at least a tag as a key=value pair to search for. Postfix the key with _bech32 to search bech32-encoded addresses or public keys"))
+		var tags []string
+		err := r.ParseForm()
+		if err != nil {
+			utils.WriteErrorResponse(w, http.StatusBadRequest, sdk.AppendMsgToErr("could not parse query parameters", err.Error()))
 			return
 		}
-
-		keyValue := strings.Split(tag, "=")
-		key := keyValue[0]
-
-		value, err := url.QueryUnescape(keyValue[1])
-		if err != nil {
-			utils.WriteErrorResponse(w, http.StatusBadRequest, sdk.AppendMsgToErr("could not decode address", err.Error()))
+		if len(r.Form) == 0 {
+			utils.PostProcessResponse(w, cdc, "[]", cliCtx.Indent)
 			return
 		}
 
-		if strings.HasSuffix(key, "_bech32") {
-			bech32address := strings.Trim(value, "'")
-			prefix := strings.Split(bech32address, "1")[0]
-			bz, err := sdk.GetFromBech32(bech32address, prefix)
+		for key, values := range r.Form {
+			value, err := url.QueryUnescape(values[0])
 			if err != nil {
-				utils.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+				utils.WriteErrorResponse(w, http.StatusBadRequest, sdk.AppendMsgToErr("could not decode query value", err.Error()))
 				return
 			}
 
-			tag = strings.TrimRight(key, "_bech32") + "='" + sdk.AccAddress(bz).String() + "'"
+			if strings.HasSuffix(key, "_bech32") {
+				prefix := strings.Split(value, "1")[0]
+				bz, err := sdk.GetFromBech32(value, prefix)
+				if err != nil {
+					utils.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+					return
+				}
+
+				key = strings.TrimRight(key, "_bech32")
+				value = sdk.AccAddress(bz).String()
```

Comment:
And then we're forcing it to an account address?

Reply:
Can you send stdtxs with validator addresses?

Reply:
We should autodetect the type using the bech prefix